spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java

```diff
@@ -107,17 +111,36 @@ public void setKeysToSanitize(String... keysToSanitize) {
 		this.sanitizer.setKeysToSanitize(keysToSanitize);
 	}
 
-	@ReadOperation
+	/**
+	 * Retrieves info for all registered configuration properties.
+	 * @return info for all registered configuration properties
+	 * @deprecated in favor of {@link #configurationProperties(String)}
+	 */
 	public ApplicationConfigurationProperties configurationProperties() {
-		return extract(this.context);
+		return configurationProperties(null);
+	}
+
+	@ReadOperation
+	public ApplicationConfigurationProperties configurationProperties(@Nullable String prefix) {
+		Predicate<ConfigurationPropertiesBean> prefixPredicate = !StringUtils.hasText(prefix) ? (bean) -> true
+				: (bean) -> bean.getAnnotation().prefix().startsWith(prefix);
+		return extract(this.context, prefixPredicate);
 	}
 
-	private ApplicationConfigurationProperties extract(ApplicationContext context) {
+	@ReadOperation
+	public ApplicationConfigurationProperties configurationPropertiesByClassName(@Selector String simpleClassName) {
```

Comment:
Also kept it constrained to a exact match on the simple class name of the backing ConfigurationProperties instance.